02/rdf/query/query-4.sparql

```diff
@@ -0,0 +1,24 @@
+PREFIX ex: <http://example.org/vocabulary/>
+PREFIX foaf: <http://xmlns.com/foaf/0.1/> 
+PREFIX dbo: <http://dbpedia.org/resource/classes#>
+
+
+
+# get streets with sidewalk and count the number of occupants living on each of those streets
+
+SELECT ?street ?streetName (COUNT(?person) AS ?count)
+WHERE {
+    ?street a dbo:Street ;
+        ex:streetName ?streetName ;
+        ex:hasSidewalk true .
+    ?building a dbo:Building ;
+        ex:situatedOn ?street .
+    OPTIONAL {
+        ?occupant a ex:Occupant ;
+            ex:livesIn ?building .
+        ?person a foaf:Person ;
```

Comment:
I think the variable `?person`  is redundant. If there is no perticular reason of leaving it there, I would delete it.